Core/Src/CAN/inverter_can_monitor.c

```diff
@@ -7,96 +7,77 @@
 
 #include "CAN/inverter_can_monitor.h"
 
+#include "CAN/inverter_can.h"
 #include "util/error_treatment.h"
 #include "util/global_definitions.h"
 #include "util/global_instances.h"
 #include "util/util.h"
 
-void set_inverter_communication_error();
 void precharge_monitor();
 void left_inv_error_callback();
 void right_inv_error_callback();
 void inverter_can_diff(uint32_t id);
 
-static bool left_inv_received = 0;
-static bool right_inv_received = 0;
-static bool is_left_inv_active = 0;
-static bool is_right_inv_active = 0;
-static FDCAN_RxHeaderTypeDef RxHeader;
-
-
 void inverter_comm_error(void* argument) {
     UNUSED(argument);
 
     for (;;) {
+
         ECU_ENABLE_BREAKPOINT_DEBUG();
 
-        uint32_t id = RxHeader.Identifier; //iniciei com esse valor pois está dando o erro " 'id' undeclared "
+        uint32_t id;
 
         switch (osMessageQueueGet(q_ids_can_inverterHandle, &id, NULL, osWaitForever)) {
 
-            default:
-                clear_error(INVERTER_COMM_ERROR_FLAG); //manter?
-                precharge_monitor();				   //manter?
+            case osOK: inverter_can_diff(id); break;
+            default: break;
```

Comment:
O que você está fazendo é isso aqui:
```suggestion
        if (osMessageQueueGet(q_ids_can_inverterHandle, &id, NULL, osWaitForever)
            == osOK) {
            inverter_can_diff(id);
        }
```
Então deixa assim com o if, fica mais claro. O switch não é para essa função.